libvast/builtins/operators/top_rare.cpp

```diff
@@ -0,0 +1,108 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/argument_parser.hpp>
+#include <vast/concept/parseable/string/char_class.hpp>
+#include <vast/concept/parseable/vast/pipeline.hpp>
+#include <vast/detail/string_literal.hpp>
+#include <vast/error.hpp>
+#include <vast/location.hpp>
+#include <vast/logger.hpp>
+#include <vast/pipeline.hpp>
+#include <vast/plugin.hpp>
+
+namespace vast::plugins::top_rare {
+
+namespace {
+
+template <detail::string_literal Name>
+// This operator does nothing but provide the operator name - the actual
+// functionality is currently implemented as a "summarize | sort" subpipeline.
+class top_rare_operator final : public crtp_operator<top_rare_operator<Name>> {
```

Comment:
This is not needed, you can delete the whole class. With the new plugin types, you inherit from `operator_parser_plugin` instead of `operator_plugin` for aliases. The difference is that one only adds an operator to the parser (as an alias), while the other adds an operator to the parser but also adds its implementation (in particular for serialization).

Reply:
Improved in https://github.com/tenzir/tenzir/pull/3176/commits/42c3e7d87b58d54f1d1e5142a0f3fa372df02848